src/party_menu.c

```diff
@@ -2985,6 +3003,42 @@ static void CB2_ReturnToPartyMenuFromSummaryScreen(void)
     InitPartyMenu(gPartyMenu.menuType, KEEP_PARTY_LAYOUT, gPartyMenu.action, TRUE, PARTY_MSG_DO_WHAT_WITH_MON, Task_TryCreateSelectionWindow, gPartyMenu.exitCallback);
 }
 
+static void CursorCb_Rename(u8 taskId)
+{
+    PlaySE(SE_SELECT);
+    sPartyMenuInternal->exitCallback = CB2_ShowRenameScreenFromPartyMenu;
+    Task_ClosePartyMenu(taskId);
+}
+
+static void CB2_ShowRenameScreenFromPartyMenu(void)
+{
+    struct Pokemon *mon = &gPlayerParty[gPartyMenu.slotId];
+    GetMonNickname(mon, gStringVar1);
+    DoNamingScreen(NAMING_SCREEN_NICKNAME, gStringVar1, GetMonData(mon, MON_DATA_SPECIES), GetMonGender(mon), GetMonData(mon, MON_DATA_PERSONALITY), CB2_ReturnToPartyMenuFromRenameScreen);
+}
+
+static void CB2_ReturnToPartyMenuFromRenameScreen(void)
+{
+    SetMonData(&gPlayerParty[gPartyMenu.slotId], MON_DATA_NICKNAME, gStringVar1);
+    gPaletteFade.bufferTransferDisabled = TRUE;
+    InitPartyMenu(gPartyMenu.menuType, KEEP_PARTY_LAYOUT, gPartyMenu.action, TRUE, PARTY_MSG_DO_WHAT_WITH_MON, Task_TryCreateSelectionWindow, gPartyMenu.exitCallback);
+}
+
+static void CursorCb_Relearn(u8 taskId)
+{
+    PlaySE(SE_SELECT);
+    gSpecialVar_0x8004 = gPartyMenu.slotId;
+    sPartyMenuInternal->exitCallback = CB2_UpdateRelearnerScreen;
+    Task_ClosePartyMenu(taskId);
+    TeachMoveRelearnerMove();
+}
+
+static void CB2_UpdateRelearnerScreen(void)
```

Comment:
it feels wrong to do this but i couldn't figure out how to make it work otherwise -- would appreciate advice